### PR TITLE
CIV-5408 Longer period of time

### PR DESCRIPTION
**Pull request by laperla11**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5408

### Change description ###
Develop the 'Add a longer period of time" on unavailable for hearing screen

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/routes/features/directionsQuestionnaire/hearing/cantAttendHearingInNext12MonthsController.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
 import {Response, Router} from 'express';
-import {DQ_NEXT_12MONTHS_CAN_NOT_HEARING_URL, UNAUTHORISED_URL} from '../../../urls';
+import {
+  DQ_NEXT_12MONTHS_CAN_NOT_HEARING_URL,
+  DQ_PHONE_OR_VIDEO_HEARING_URL,
+  DQ_AVAILABILITY_DATES_FOR_HEARING_URL,
+} from '../../../urls';
 import {GenericForm} from '../../../../common/form/models/genericForm';
```
Comment (laarmada) [inline src/main/routes/features/directionsQuestionnaire/hearing/cantAttendHearingInNext12MonthsController.ts:7]: we could try to cut the import url when we are modify this file....

File: src/main/routes/features/directionsQuestionnaire/witnesses/otherWitnessesController.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import {NextFunction, Response, Router} from 'express';
 import {GenericForm} from '../../../../common/form/models/genericForm';
-import {DQ_DEFENDANT_WITNESSES_URL, DQ_OTHER_WITNESSES_AVAILABILITY_DATES_FOR_HEARING_URL} from '../../../urls';
+import {
+  DQ_DEFENDANT_WITNESSES_URL,
+  DQ_NEXT_12MONTHS_CAN_NOT_HEARING_URL,
+} from '../../../urls';
 import {OtherWitnesses} from '../../../../common/models/directionsQuestionnaire/witnesses/otherWitnesses';
```
Comment (laarmada) [inline src/main/routes/features/directionsQuestionnaire/witnesses/otherWitnessesController.ts:7]: same here about cut urls

File: src/main/services/features/directionsQuestionnaire/hearing/unavailableDatesForHearingService.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {UnavailableDatePeriod, UnavailableDates, UnavailableDateType} from 'common/models/directionsQuestionnaire/hearing/unavailableDates';
+
+export const getUnavailableDatesForm = (reqBody: Record<string, []>): UnavailableDates => {
+  const unavailableDates = reqBody.items.map((item: any) => {
+    if (item.type === UnavailableDateType.SINGLE_DATE) {
```
Comment (laarmada) [inline src/main/services/features/directionsQuestionnaire/hearing/unavailableDatesForHearingService.ts:5]: should we replace all that code:
 if (item.type === UnavailableDateType.SINGLE_DATE) {
return new UnavailableDatePeriod(UnavailableDateType.SINGLE_DATE, item.single.start);
    }
    if (item.type === UnavailableDateType.LONGER_PERIOD) {
      return new UnavailableDatePeriod(UnavailableDateType.LONGER_PERIOD, item.period.start, item.period.end);
    }

for: 
 if (item?.type) {
    return new UnavailableDatePeriod(item.type, item.period.start, item.period.end);
 }

**Review by laarmada — CHANGES_REQUESTED** [review]

**Comment by laarmada** [comment]
really nice that idea:
@MaxDate(addMonths(new Date(), 12), { message: generateErrorMessageForMaxDate })

**Review by laarmada — DISMISSED** [review]

File: src/main/services/features/directionsQuestionnaire/hearing/unavailableDatesForHearingService.ts (hunk 3 above)
Comment (laarmada) [reply src/main/services/features/directionsQuestionnaire/hearing/unavailableDatesForHearingService.ts:5]: we will discuss offline

**Review by TatsyMaslova — APPROVED** [review]

**Review by portizdev — DISMISSED** [review]

**Review by pascalkankonde — APPROVED** [review]

**Review by laarmada — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]
LGTM

**Pull request merged**